src/app/gps/qgsappgpsdigitizing.cpp

```diff
@@ -97,6 +102,26 @@ QgsAppGpsDigitizing::~QgsAppGpsDigitizing()
   mRubberBand = nullptr;
 }
 
+double QgsAppGpsDigitizing::totalTrackLength() const
+{
+  QVector<QgsPointXY> points;
+  QgsGeometry::convertPointList( mCaptureListWgs84, points );
```

Comment:
Why discarding elevation when calculating distance here?

Reply:
We don't do ellipsoidal measurements with elevation (yet?)

Reply:
Might be worth noting somewhere that's it's a flattened length maths with a Todo: calculate 3d length whenever we can :)

Reply:
I'm just not sure if that's even possible -- I haven't found any libraries which offer 3d ellipsoidal calculations to date. 😱